backend/src/test/java/ai/verta/modeldb/GlobalSharingTest.java

```diff
@@ -0,0 +1,451 @@
+      String authServiceHost = (String) authServicePropMap.get(ModelDBConstants.HOST);
+      Integer authServicePort = (Integer) authServicePropMap.get(ModelDBConstants.PORT);
+      app.setAuthServerHost(authServiceHost);
+      app.setAuthServerPort(authServicePort);
+
+      authService = new AuthServiceUtils();
+      roleService = new RoleServiceUtils(authService);
+    }
+
+    ModelDBHibernateUtil.runLiquibaseMigration(databasePropMap);
+    App.initializeServicesBaseOnDataBase(
+        serverBuilder, databasePropMap, propertiesMap, authService, roleService);
+    serverBuilder.intercept(new AuthInterceptor());
+
+    Map<String, Object> testUerPropMap = (Map<String, Object>) testPropMap.get("testUsers");
+    if (testUerPropMap != null && testUerPropMap.size() > 0) {
+      authClientInterceptor = new AuthClientInterceptor(testPropMap);
+      client1ChannelBuilder.intercept(authClientInterceptor.getClient1AuthInterceptor());
+      client2ChannelBuilder.intercept(authClientInterceptor.getClient2AuthInterceptor());
+    }
+
+    if (app.getAuthServerHost() != null && app.getAuthServerPort() != null) {
+      ManagedChannel authServiceChannel =
+          ManagedChannelBuilder.forTarget(app.getAuthServerHost() + ":" + app.getAuthServerPort())
+              .usePlaintext()
+              .intercept(authClientInterceptor.getClient1AuthInterceptor())
+              .build();
+      // all service stubs
+      organizationServiceBlockingStub = OrganizationServiceGrpc.newBlockingStub(authServiceChannel);
+      roleServiceBlockingStub = RoleServiceGrpc.newBlockingStub(authServiceChannel);
+      collaboratorServiceStub = CollaboratorServiceGrpc.newBlockingStub(authServiceChannel);
+    }
+
+    serverBuilder.build().start();
+    ManagedChannel channel = client1ChannelBuilder.maxInboundMessageSize(1024).build();
+    ManagedChannel client2Channel = client2ChannelBuilder.maxInboundMessageSize(1024).build();
+    deleteEntitiesCron = new DeleteEntitiesCron(authService, roleService, 1000);
+
+    // Create all service blocking stub
+    projectServiceStub = ProjectServiceGrpc.newBlockingStub(channel);
+    client2ProjectServiceStub = ProjectServiceGrpc.newBlockingStub(client2Channel);
+    datasetServiceStub = DatasetServiceGrpc.newBlockingStub(channel);
+    client2DatasetServiceStub = DatasetServiceGrpc.newBlockingStub(client2Channel);
+    repositoryServiceStub = VersioningServiceGrpc.newBlockingStub(channel);
+    client2RepositoryServiceStub = VersioningServiceGrpc.newBlockingStub(client2Channel);
+  }
+
+  @AfterClass
+  public static void removeServerAndService() {
+    App.initiateShutdown(0);
+
+    // Remove all entities
+    // removeEntities();
+    // Delete entities by cron job
+    deleteEntitiesCron.run();
+
+    // shutdown test server
+    serverBuilder.build().shutdownNow();
+  }
+
+  private static final Collection<Object[]> requestData =
+      Arrays.asList(
+          new Object[][] {
```

Comment:
Could we define an actual type? I'm not sure what each field means and I really like relevant names for things.

Reply:
added type for that

Reply:
Thanks!